pkg/investigations/investigation.go

```diff
@@ -9,23 +9,36 @@ import (
 	hivev1 "github.com/openshift/hive/apis/hive/v1"
 )
 
+type InvestigationStep struct {
+	Performed bool
+	Labels    []string
+}
+
+type InvestigationResult struct {
+	InvestigationName  string
```

Comment:
We should know the investigation name after calling `GetInvestigation` in `investigate.go`.

Would it be more logical to just pass that to `updateMetrics` instead of having it as pat of the `InvestigationResult`? That way, we can also remove the local variables for the investigation names in the investigation specific files, as those have no good reason to be there.